src/lib/mdns/tests/TestTxtFields.cpp

```diff
@@ -372,35 +381,146 @@ void TestFillNodeDataFromTxt(nlTestSuite * inSuite, void * inContext)
+    FillNodeDataFromTxt(GetSpan(key), GetSpan(val), nodeData);
+    NL_TEST_ASSERT(inSuite, nodeData.mMrpRetryIntervalIdle == 1);
+
+    // Maximum
+    sprintf(key, "CRI");
+    sprintf(val, "3600000");
+    FillNodeDataFromTxt(GetSpan(key), GetSpan(val), nodeData);
+    NL_TEST_ASSERT(inSuite, nodeData.mMrpRetryIntervalIdle == 3600000);
+
+    // Test no other fields were populated
+    nodeData.mMrpRetryIntervalIdle = 0;
+    NL_TEST_ASSERT(inSuite, NodeDataIsEmpty(nodeData));
+
+    // Invalid CRI => fallback to 0
+    sprintf(key, "CRI");
+    sprintf(val, "-1");
+    FillNodeDataFromTxt(GetSpan(key), GetSpan(val), nodeData);
+    NL_TEST_ASSERT(inSuite, nodeData.mMrpRetryIntervalIdle == 0);
+
+    // Invalid CRI => fallback to 0
+    sprintf(key, "CRI");
+    sprintf(val, "asdf");
+    FillNodeDataFromTxt(GetSpan(key), GetSpan(val), nodeData);
+    NL_TEST_ASSERT(inSuite, nodeData.mMrpRetryIntervalIdle == 0);
+}
+
+// Test CRA
+void TxtFieldMrpRetryIntervalActive(nlTestSuite * inSuite, void * inContext)
+{
+    char key[4];
+    char val[8];
+    ResolvedNodeData nodeData;
+
+    // Minimum
+    sprintf(key, "CRA");
+    sprintf(val, "1");
+    FillNodeDataFromTxt(GetSpan(key), GetSpan(val), nodeData);
+    NL_TEST_ASSERT(inSuite, nodeData.mMrpRetryIntervalActive == 1);
+
+    // Maximum
+    sprintf(key, "CRA");
+    sprintf(val, "3600000");
+    FillNodeDataFromTxt(GetSpan(key), GetSpan(val), nodeData);
+    NL_TEST_ASSERT(inSuite, nodeData.mMrpRetryIntervalActive == 3600000);
+
+    // Test no other fields were populated
+    nodeData.mMrpRetryIntervalActive = 0;
+    NL_TEST_ASSERT(inSuite, NodeDataIsEmpty(nodeData));
+
+    // Invalid CRI => fallback to 0
+    sprintf(key, "CRA");
+    sprintf(val, "-1");
+    FillNodeDataFromTxt(GetSpan(key), GetSpan(val), nodeData);
+    NL_TEST_ASSERT(inSuite, nodeData.mMrpRetryIntervalActive == 0);
+
+    // Invalid CRI => fallback to 0
+    sprintf(key, "CRA");
+    sprintf(val, "asdf");
+    FillNodeDataFromTxt(GetSpan(key), GetSpan(val), nodeData);
+    NL_TEST_ASSERT(inSuite, nodeData.mMrpRetryIntervalActive == 0);
+}
+
+// Test T (TCP support)
+void TxtFieldTcpSupport(nlTestSuite * inSuite, void * inContext)
+{
+    char key[4];
+    char val[8];
+    ResolvedNodeData nodeData;
+
+    // True
+    sprintf(key, "T");
+    sprintf(val, "1");
+    FillNodeDataFromTxt(GetSpan(key), GetSpan(val), nodeData);
+    NL_TEST_ASSERT(inSuite, nodeData.mSupportsTcp);
+
+    // Test no other fields were populated
+    nodeData.mSupportsTcp = false;
+    NL_TEST_ASSERT(inSuite, NodeDataIsEmpty(nodeData));
+
+    // False
+    sprintf(key, "T");
+    sprintf(val, "0");
+    FillNodeDataFromTxt(GetSpan(key), GetSpan(val), nodeData);
+    NL_TEST_ASSERT(inSuite, !nodeData.mSupportsTcp);
+
+    // Invalid value, stil false
+    sprintf(key, "T");
+    sprintf(val, "asdf");
+    FillNodeDataFromTxt(GetSpan(key), GetSpan(val), nodeData);
+    NL_TEST_ASSERT(inSuite, !nodeData.mSupportsTcp);
```

Comment:
Spec says:

> If the key is not included or invalid, the Node querying the service record SHALL expect the default value of T=1 indicating TCP is supported. 

as far as I can tell...

Reply:
Good point. I kind of thought it's obvious the default should be "no TCP support" :)

Reply:
Yeah, I'm not sure why the spec says what it does here, and maybe that should be raised as a spec issue instead....  If we do that, then implementing the non-spec behavior, with a pointer to the issue, is probably fine.

Reply:
Looks like the spec here changed today (yesterday?) in https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/3647?  @Damian-Nordic